src/include/optimizer/rule_rewrite.h

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rule_rewrite.h
+//
+// Identification: src/include/optimizer/rule_rewrite.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "optimizer/rule.h"
+#include "optimizer/absexpr_expression.h"
+
+#include <memory>
+
+namespace peloton {
+namespace optimizer {
+
+/* Rules are applied from high to low priority */
+enum class RulePriority : int {
+  HIGH = 3,
+  MEDIUM = 2,
+  LOW = 1
+};
+
+class ComparatorElimination: public Rule<AbsExpr_Container,ExpressionType,AbsExpr_Expression> {
```

Comment:
It would be nice if you added some documentation to this, even if it's simplistic.